File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    
    if s[0] == ')' or s[-1] == '(': #우선 앞이 ) , 뒤가 (로 끝나면 안됨 false
        return False
    else:
        stack = list(s)
    
    
    temp = 0
    for data in stack:
        if data == '(':
            temp += 1
        else:
            temp -= 1
        
        if temp < 0: #중간에 -1 이 나오면 바로 false
            return False
    
    if temp == 0: #최종 합이 0이어야 true
        return True
    else:
        return False
        

    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var stack = [Character]()
    
    for c in s {
        if c == "(" {
            stack.append("(")
        } else {
            guard stack.last == "(" else { return false }
            stack.popLast()
        }
    }
    
    return stack.isEmpty ? true : false
}